__init__.py

```diff
@@ -64,11 +66,44 @@ def __init__(self):
         self.weather_api = OpenWeatherMapApi()
         self.weather_api.set_language_parameter(self.lang)
         self.platform = self.config_core["enclosure"].get("platform", "unknown")
+        self.gui_image_directory = Path(self.root_dir).joinpath("ui")
         self.weather_config = None
 
     def initialize(self):
         """Do these things after the skill is loaded."""
         self.weather_config = WeatherConfig(self.config_core, self.settings)
+        self.add_event("skill.weather.request-local", self.handle_get_local_weather)
+
+    def handle_get_local_weather(self, _):
+        """Handles a message bus command requesting current local weather information.
+
+        Such a request will typically come from a domain external to this skill that
+        requires weather information but should not go through the intent system
+        to get it.
+        """
+        system_unit = self.config_core.get("system_unit")
+        try:
+            weather = self.weather_api.get_weather_for_coordinates(
+                system_unit, self.weather_config.latitude, self.weather_config.longitude
+            )
+        except Exception:
+            self.log.exception("Unexpected error getting weather for skill API.")
```

Comment:
Do we want to respond with anything to the request? Or just let the request timeout and it will retry?

If we're continually getting a failure, do we just keep retrying indefinitely?

Reply:
Again, I am trying to avoid using the request/response mechanism.  The `skill.weather.local-retrieved` event should only emit when the API call was successful (otherwise it is lying).  If we want a `skill.weather.local-failed`, or something of the like, we can add it, but I can't think of anything the home screen skill would do with it.